samples/nfc/system_off/src/main.c

```diff
@@ -0,0 +1,187 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <stdio.h>
+#include <zephyr.h>
+#include <power/power.h>
+
+#include <nrfx.h>
+#include <hal/nrf_power.h>
+
+#include <nfc_t2t_lib.h>
+#include <nfc/ndef/nfc_ndef_msg.h>
+#include <nfc/ndef/nfc_text_rec.h>
+
+#include <dk_buttons_and_leds.h>
+
+
+#define SYSTEM_OFF_DELAY_S	3
+
+#define MAX_REC_COUNT		1
+#define NDEF_MSG_BUF_SIZE	128
+
+#define NFC_FIELD_LED		DK_LED1
+#define SYSTEM_ON_LED		DK_LED2
+
+
+/* Delayed work that enters system off. */
+static struct k_delayed_work system_off_work;
+
+
+/**
+ * @brief Function that receives events from NFC.
+ */
+static void nfc_callback(void *context,
+			 enum nfc_t2t_event event,
+			 const u8_t *data,
+			 size_t data_length)
+{
+	ARG_UNUSED(context);
+	ARG_UNUSED(data);
+	ARG_UNUSED(data_length);
+
+	switch (event) {
+	case NFC_T2T_EVENT_FIELD_ON:
+		/* Cancel entering system off */
+		k_delayed_work_cancel(&system_off_work);
+		dk_set_led_on(NFC_FIELD_LED);
+		break;
+	case NFC_T2T_EVENT_FIELD_OFF:
+		/* Enter system off after delay */
+		k_delayed_work_submit(&system_off_work,
+				K_SECONDS(SYSTEM_OFF_DELAY_S));
+		dk_set_led_off(NFC_FIELD_LED);
+		break;
+	default:
+		break;
+	}
+}
+
+
+/**
+ * @brief Function for configuring and starting the NFC.
+ */
+static int start_nfc(void)
```

Comment:
Did you consider having NFC tag without actual NDEF data in this sample (like in the nRF5 SDK example)? In this case, you would just need NFCT HAL from nrfx to set up the peripheral in Sense mode so that it is possible to wake it up using NFC. 

I think it would be valuable for demonstration purposes to have minimum code that shows how to wake up the device from System Off using NFC. The NFC library (with NDEF message) is a nice addition that makes the sample more fun and interactive, but it might make users think that NFC library is hard requirement for this functionality.

It is just my opinion and I'm curious what others think about it.

Reply:
There are reasons why I added NDEF:
1. Example is more interactive as you point out. It gives more feedback to user.
2. It coves more common use - if user wants wake up from NFC field then he probably wants actual NFC tag working.
3. I agree that users may think that NFC library is hard requirement, but on the other hand if we show direct use of NFCT HAL from nrfx then user may think that he have to use it always to make this feature working when actually it is very simple with NFC library.

Reply:
In the nRF5 SDK we used to have this minimal version of this sample, where NFCT HAL was used directly to set up NFCT as wakeup source. I don't have a strong opinion on this, I think both approaches have their advantages. Maybe as a compromise, please add a note in the documentation, that when NFC tag library is not needed, and the NFCT is only used as a wakeup source for the system, the NFCT HAL functions can be used directly to setup NFCT as weakup. BTW doesn't Zephyr's power mode interface have a dedicated API for setting wakeup source?

Reply:
1. I added `NFCT only` description to the documentation. @greg-fer please review latest changes in README:
https://github.com/NordicPlayground/fw-nrfconnect-nrf/compare/b72773e7..50d24730

2. No, I don't see such API.